src/sync/generic_network_transport.hpp

```diff
@@ -46,8 +46,9 @@ enum class JSONErrorCode {
     missing_json_key = 3,
 };
 
-const std::error_category& json_error_category() noexcept;
-std::error_code make_error_code(JSONErrorCode) noexcept;
+enum class BSONErrorCode {
```

Comment:
Why not add this value as a new value inside of the existing `JSONErrorCode`? (non-blocking)

Reply:
Your suggestion is a lot cleaner than having a seperate enum. I've added the error state to JSONErrorCode